### ISSUE TITLE
Config: wrong *format entries +1

### ISSUE THREAD

**Issue by rurban** [perl11/cperl#61]
wrong length with entries ending with ", caused by calculating the strlen with \" as two chars. unquote it.

```
$ cperl -V:uvuformat
uvuformat='"lu"m';
$ cperl -V:ivdformat
uvuformat='"ld"n;
$ cperl -V:nvgformat
uvuformat='"g"b';
```

previously also seen with ending \0 byte until we changed to U16 keys.

**Comment by bulk88** [comment]
This commit https://github.com/perl11/cperl/commit/1097d2a070c32a3920105fba04fa7af2c1049d33 is faulty makes it impossible to compile any XS code on Win32 Cperl since some Config entries are truncated, it all fails with "LINK : fatal error LNK1146: no argument specified with option '/machine:'" since the last 3 chars are being cut off the XS Config entry. I think instead of '$qv =~ s/\//g;' the config string  needs to be evaled (slightly dangerous with regards to future proofing) or parsed by a proper C-like escape sequence processor.

example

```
C:\sources\cperl\ext\Config>..\..\cperl -I..\..\lib -MConfig -E"say $Config{lddl
flags}"
-dll -nologo -nodefaultlib -debug -opt:ref,icf -ltcg            -libpath:"c:\per
l\lib\CORE"             -machine:
```

from Config.xs

```
#line 947 "ext/Config/Config_xs.in"
      {(U16)(long)&((struct stringpool_t *)0)->stringpool_str2499,      T_STR,  94,"-dll -nologo -nodefaultlib -debug -opt:ref,icf -ltcg        -libpath:\"c:\\perl\\lib\\CORE\"        -machine:x86"}, /* lddlflags */
      {U16_MAX, -1, 0, NULL},
```

Failures from smoke

```
Test Summary Report
-------------------
comp/require.t                                                  (Wstat: 0 Tests
 58 Failed: 1)
  Failed test:  53
../cpan/ExtUtils-Constant/t/Constant.t                          (Wstat: 0 Tests
 3 Failed: 1)
  Failed test:  3
  Parse errors: Bad plan.  You planned 302 tests but ran 3.
../cpan/ExtUtils-Install/t/Installed.t                          (Wstat: 1280 Te
ts: 73 Failed: 5)
  Failed tests:  56, 64-67
  Non-zero exit status: 5
../cpan/ExtUtils-MakeMaker/t/xs.t                               (Wstat: 256 Tes
s: 5 Failed: 1)
  Failed test:  4
  Non-zero exit status: 1
../dist/ExtUtils-ParseXS/t/001-basic.t                          (Wstat: 512 Tes
s: 8 Failed: 2)
  Failed tests:  7-8
  Non-zero exit status: 2
  Parse errors: Bad plan.  You planned 17 tests but ran 8.
../dist/ExtUtils-ParseXS/t/002-more.t                           (Wstat: 512 Tes
s: 9 Failed: 3)
  Failed tests:  6-7, 9
  Non-zero exit status: 2
  Parse errors: Bad plan.  You planned 29 tests but ran 9.
../dist/ExtUtils-ParseXS/t/003-usage.t                          (Wstat: 3072 Te
ts: 24 Failed: 12)
  Failed tests:  6-7, 9, 11, 13, 15, 17-18, 20-22, 24
  Non-zero exit status: 12
Files=2396, Tests=705412, 2229 wallclock secs (80.50 usr +  3.31 sys = 83.81 CP
)
```

**Comment by rurban** [comment]
Uh, windows paths!
I didn't check for octal values escaped, like \000 neither.

**Comment by rurban** [comment]
I think I'll rather use sizeof(key) here.

**Comment by rurban** [comment]
Fixed with GH #64
